src/monattack.cpp

```diff
@@ -185,6 +185,94 @@ void mattack::shriek(monster *z, int index)
     sounds::sound(z->pos(), 50, _("a terrible shriek!"));
 }
 
+void mattack::shriek_alert(monster *z, int index)
+{
+    if( !z->can_act() || z->has_effect("shrieking")) {
+        return;
+    }
+
+    Creature *target = z->attack_target();
+
+    if(g->u.sees( *z )){
+    add_msg( _("The %s begins shrieking!"), z->name().c_str());
+    }
+
+    int dist;
+    if( target == nullptr || (dist = rl_dist( z->pos(), target->pos() )) > 15 ||
+        !z->sees( *target ) ) {
+        return;
+    }
+
+    z->moves -= 150;
+    z->reset_special(index); // Reset timer
+    sounds::sound(z->pos(), 120, _("a piercing wail!"));
+    z->add_effect("shrieking", 10);
+
+}
+
+static int posp( int posc ) //Determines orientation of target from the source
+{
+    if( posc > 0 ) {
+        return 1;
+    }
+    if( posc < 0 ) {
+        return -1;
+    } else {
+        return 0;
+    }
+}
+
+void mattack::shriek_stun(monster *z, int index)
+{
+    if( !z->can_act() || !z->has_effect("shrieking")) {
+        return;
+    }
+
+    Creature *target = z->attack_target();
+    int dist;
+    if( target == nullptr || (dist = rl_dist( z->pos(), target->pos() )) > 7 ||
+        !z->sees( *target ) ) {
+        return;
+    }
+
+    z->reset_special(index); // Reset timer
+
+    float dx = posp( target->posx() - z->posx() );
+    float dy = posp( target->posy() - z->posy() ); //Find direction of player
+    int i = 0; int j = 0;
+    int bsize = 0;  //Set range of our boxes
+    for( int range = 0; range < 4; range++ ) {
+        dx = dx * 1.5;
+        dy = dy * 1.5;
+        i++;  j++; bsize--; //Increase size of boxes after each distance toward target
+        for (int boxx = bsize; boxx <= i; boxx++) {
+            for (int boxy = bsize; boxy <= j; boxy++) {
+                tripoint cone( z->posx() + dx + boxx,
+                               z->posy() + dy + boxy,
+                               z->posz() );
+            if( z->sees(cone) ) {
+                g->m.bash( cone, 3, true ); //Small bash to every square, silent to not flood message box
+                if (g->mon_at(cone) != -1){
+                    const int mondex = g->mon_at(cone);
+                    Creature *zed = &g->zombie( mondex ); //If a monster is there, chance for stun
+                    if ( one_in((dist + 8 )) && !(zed->is_immune_effect("deaf")) && !zed->has_effect("stunned") ){
+                        zed->add_effect("stunned", rng(2,3));
+
+                    }
+                } else if (cone == g->u.pos() || cone == target->pos()){ //Same for player
+                    if ( one_in((dist + 8 )) && !(target->is_immune_effect("deaf")) && !target->has_effect("stunned") ){
+                        target->add_effect("stunned", rng(2,3));
```

Comment:
> 'dazed'

No offense taken, I just have no idea how to spell :P



> would kinda duplicate what boomers do.

True.. In the interest of creating new interesting things, how about it also increasing the dispersion of all ranged weapons you use? So screechers would alert nearby enemies of you, then decrease your overall fighting potential, lowering your speed and making it harder to aim.